### PR TITLE
feat(tests): Add integration tests for policy store loader

### LINKED ISSUE #12881: feat(jans-cedarling): Add end‑to‑end tests for new policy store loader in cedarling/src/tests
**Is your feature request related to a problem? Please describe.**
add new integration tests under cedarling/src/tests to verify that the new policy store loader and manifest-based policy store format work correctly end‑to‑end. The goal is to move beyond unit tests in common::policy_store and assert real-world behavior through the same Cedarling API used by existing authorization tests.

**Describe the solution you'd like**
Add tests following the same pattern as the other tests

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by haileyesus2433**
- Introduced a new test module `policy_store_loader.rs` to validate the functionality of the policy store loader.
- Implemented tests for loading policy stores from both directory structures and Cedar Archive (.cjar) files, ensuring correct authorization behavior.
- Added manifest validation tests to check for checksum mismatches and policy store ID mismatches.
- Enhanced existing test utilities to support the new loader functionality.

### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Target issue
closes #12881 

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/93cef6c2-2455-4450-a780-671611c818e8?refs=8dfa1254e38d9818d6d13e1b1d81b38b6241c5b3&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs [hunk 1]
```diff
@@ -0,0 +1,1003 @@
...
+
+/// Test that manifest validation detects checksum mismatches.
+///
+/// This test uses `load_policy_store_directory` which performs manifest validation.
+/// An invalid checksum format in the manifest should cause initialization to fail.
+#[test]
+#[cfg(not(target_arch = "wasm32"))]
+async fn test_manifest_validation_invalid_checksum_format() {
+    use crate::common::policy_store::test_utils::fixtures;
+    use crate::{BootstrapConfig, LogConfig, LogTypeConfig, PolicyStoreConfig};
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: Is there a reason for using imports per test?

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs [hunk 2]
```diff
@@ -0,0 +1,1003 @@
...
+    })
+    .await
+}
+
+/// Creates a Cedarling instance from an archive file path.
+///
+/// Disables default entity building (user, workload, roles) since we're using
+/// a custom schema that doesn't include the Jans namespace types.
+/// Uses a custom principal_bool_operator that checks for TestApp::User principal.
+#[allow(dead_code)]
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: Remove this `#[allow(dead_code)]` since the function is being used

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 1 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: yeah it is to avoid unused import warnings for WASM builds  When tests are marked with #[cfg(not(target_arch = "wasm32"))], moving the imports inside those functions ensures they only compile when the test compiles. Otherwise we would get "unused import" warnings when building for WASM.

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 2 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: sorry must have forgot this resolved in [6c86e4c](https://github.com/JanssenProject/jans/pull/12884/commits/6c86e4c172fcecadca849bb2d604f1d371c015a2)

**Comment by haileyesus2433** [comment]
> Maybe we also should check other methods that `authorize_unsigned`, just to be sure that it works as expected?

resloved in [9daf712](https://github.com/JanssenProject/jans/pull/12884/commits/9daf712cb8f55a931a0fe4b41352da5848072fdf)

**Comment by haileyesus2433** [comment]
> Have you also tested with an empty policy store directory or maybe missing required files or directories?

Already covered at unit level in loader_tests.rs

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs [hunk 3]
```diff
@@ -0,0 +1,1252 @@
...
+        policy_store_config: PolicyStoreConfig {
+            source: PolicyStoreSource::Directory(temp_dir.path().to_path_buf()),
+        },
+        jwt_config: crate::JwtConfig::new_without_validation(),
+        authorization_config: Default::default(),
+        entity_builder_config: Default::default(),
+        lock_config: None,
+        max_default_entities: None,
+        max_base64_size: None,
+    };
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: This part repeats multiple times maybe create a helper function or better yet use the `get_config` function defined in `tests::utils::cedarling_util` if it doesn't break anything

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs [hunk 4]
```diff
@@ -0,0 +1,1252 @@
...
+            ..Default::default()
+        },
+        lock_config: None,
+        max_default_entities: None,
+        max_base64_size: None,
+    };
+
+    let cedarling = crate::Cedarling::new(&config)
+        .await
+        .expect("Cedarling should initialize with JWT-enabled config");
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/tests/policy_store_loader.rs:1175]: You can use the `get_cedarling_with_authorization_conf` function with this one.

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 3 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: The manifest validation tests intentionally create BootstrapConfig directly because they test error paths  they expect Cedarling::new() to fail. The existing helpers all use .expect() which would panic instead of letting us verify the error.

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 4 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs:1175]: This test cannot use `get_cedarling_with_authorization_conf` because It needs `jwt_sig_validation: true` for actual JWT signature validation with MockServer, All existing helpers use` JwtConfig::new_without_validation()` which disables JWT validation and the test specifically validates that signed JWTs work with the policy store loader
However, it COULD `use get_cedarling_with_callback` with a callback that overrides the jwt_config but
using get_cedarling_with_callback would require overriding jwt_config, authorization_config, AND entity_builder_config in the callback  at that point it's arguably less readable than constructing the config directly.

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 3 above)
Comment (dagregi) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: I saw the test checks the validity of manifest.json in the archive so the bootstrap config got from `get_config` wouldn't change the core test as long as you pass the same policy store source, the only notable difference is the `LogConfig` which I hope doesn't affect much.

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 4 above)
Comment (dagregi) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs:1175]: I tested it locally with `jwt_sig_validation` set to false and the test passed, even using `get_cedarling_with_authorization_conf` the test still passes, maybe I missed something else so I will check again

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 3 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs]: ok fine [6002736](https://github.com/JanssenProject/jans/pull/12884/commits/6002736e63bc9e4ba63fb986eca29233622b1f0f)

File: jans-cedarling/cedarling/src/tests/policy_store_loader.rs (hunk 4 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/tests/policy_store_loader.rs:1175]: yes you missed something the purpose of the test , test passes with `jwt_sig_validation: true` because the tokens are valid and properly signed and disabling validation just skips the signature check, it doesn't reject valid/invalid tokens. To prove validation actually matters, I added an assertion with a tampered token that fails when validation is enabled and using the helpers (which disables validation) would defeat the purpose of the test since tampered tokens would be accepted. Testing with validation enabled is important because in production, JWT signature validation is a critical security control and we need to verify that Cedarling correctly rejects tampered or forged tokens when configured to do so.

here is the added assertion [8dfa125](https://github.com/JanssenProject/jans/pull/12884/commits/8dfa1254e38d9818d6d13e1b1d81b38b6241c5b3)

**Review by dagregi — APPROVED** [review]

**Review by olehbozhok — APPROVED** [review]
